blues/sparrow-starter
# feat: pir motion data

# Problem Context

The Sparrow Reference Sensor board includes a PIR sensor. This produces events in the #motion.qo file showing the number of times motion has been detected since the previous event.

## Changes

Added functions in the NotehubDataProvider to combine multiple sensor events in the `getLatestEvents` call into one object containing both `air.qo` and `motion.qo` data when present.

Added new sensor readings and schema and formatters for the `count` and `total` data properties we're now accepting and passing to the front end.

Added a new bar chart component for displaying motion count events in the sensor details page. Added a new sensor reading of `Motion` to the latest sensor events readings, and a tooltip of `Total motions detected` when it is hovered over.

Added and updated unit tests to reflect the new changes to our service layer and our new charts.

## Screenshot (if applicable)

https://user-images.githubusercontent.com/20400845/154728853-937e6107-e9ad-42f9-aa3c-c0feeb262d7f.mov

## Testing

Unit / integration / e2e tests?

Broken unit tests fixed, addt unit tests added. E2E tests in the process of being updated to account for new visuals.

Steps to test manually?

1. Go to the Sparrow home page
2. See that the sensor card now has an item for motion included along with temp, humidity, voltage and pressure
3. Click on the card and wait for details page to load
4. See the newly added motion column. Hover over it to see the total motion tooltip
5. Scroll down to see the new motion count table, hover it to see the tooltips of motion count.

Any other sorts of testing notes to include?

## Any other related PRs

n/a

## Ticket(s)

https://www.pivotaltracker.com/story/show/181272641
